scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua

```diff
@@ -28,210 +39,144 @@ end;
+            end
+        end
+    elseif(mobHPP <= 40 and change == 9) then -- Normal Form
+        mob:setModelId(1839);
+        mob:setLocalVar("change", 10);
+            if (TP <= 4) then
+            mob:useMobAbility(1860);
+            mob:setLocalVar("TP", 5)
+            end
+    elseif(mobHPP <= 50 and change == 8) then -- Cerb and Co.
+        mob:setModelId(1793);
+        mob:setLocalVar("change", 9);
+        for i = 1, 8 do
+            GetMobByID(petIDs[i]):setModelId(281);
+            if petStatus[i] == 0 then
+               SpawnMob(petIDs[i]):updateEnmity(target);
+            end
+        end
+    elseif(mobHPP <= 52 and change == 7) then -- Normal Form
+        mob:setModelId(1839);
+        mob:setLocalVar("change", 8);
+            if (TP <= 3) then
+            mob:useMobAbility(1861);
+            mob:setLocalVar("TP", 4)
+            end
+    elseif(mobHPP <= 62 and change == 6) then -- Troll and Co.
+        mob:setModelId(1867);
+        mob:setLocalVar("change", 7);
+        for i = 1, 8 do
+            GetMobByID(petIDs[i]):setModelId(1680);
+            if petStatus[i] == 0 then
+               SpawnMob(petIDs[i]):updateEnmity(target);
+            end
+        end
+    elseif(mobHPP <= 64 and change == 5) then -- Normal Form
+        mob:setModelId(1839);
+        mob:setLocalVar("change", 6);
+            if (TP <= 2) then
+            mob:useMobAbility(1862);
+            mob:setLocalVar("TP", 3)
+            end
+    elseif(mobHPP <= 74 and change == 4) then -- Lamia and Co.
+        mob:setModelId(1865);
+        mob:setLocalVar("change", 5);
+        for i = 1, 8 do
+            GetMobByID(petIDs[i]):setModelId(1643);
+            if petStatus[i] == 0 then
+               SpawnMob(petIDs[i]):updateEnmity(target);
+            end
+        end
+    elseif(mobHPP <= 76 and change == 3) then -- Normal Form
+        mob:setModelId(1839);
+        mob:setLocalVar("change", 4);
+            if (TP <= 1) then
+            mob:useMobAbility(1863);
+            mob:setLocalVar("TP", 2)
+            end
+    elseif(mobHPP <= 86 and change == 2) then -- Mamool and Co.
+        mob:setModelId(1863);
+        mob:setLocalVar("change", 3);
+        for i = 1, 8 do
+            GetMobByID(petIDs[i]):setModelId(1639);
+            if petStatus[i] == 0 then
+               SpawnMob(petIDs[i]):updateEnmity(target);
+            end
+        end
+    elseif(mobHPP <= 88 and change == 1) then -- Normal Form
+        mob:setModelId(1839);
+        mob:setLocalVar("change", 2);
+            if (TP <= 0) then
+            mob:useMobAbility(1857);
+            mob:setLocalVar("TP", 1)
+            end
+    elseif(mobHPP <= 98 and change == 0) then -- Chariots
+        mob:setModelId(1825);
+        mob:setLocalVar("change", 1);
+        for i = 1, 8 do
+            GetMobByID(petIDs[i]):setModelId(1820);
+            if petStatus[i] == 0 then
+               SpawnMob(petIDs[i]):updateEnmity(target);
+            end
+        end

```

Comment:
strip out all the for loops inside each elseif, and set a var 'skin' to whatever you wanna change the pets' modelids to
then, outside the if/elseif block loop over the petids once and set the modelid tp 'skin' (whatever it was set in the if/elseif block)